lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -186,10 +186,20 @@ def write_ssh_key
       def start_ansible_runner
         env = {}
         env['FOREMAN_CALLBACK_DISABLE'] = '1' if @rex_command
-        command = [env, 'ansible-runner', 'run', @root, '-p', 'playbook.yml']
+        command = ['ansible-runner', 'run', @root, '-p', 'playbook.yml']
         command << '--cmdline' << cmdline unless cmdline.nil?
         command << verbosity if verbose?
-        initialize_command(*command)
+
+        ansible_env = Proxy::Ansible::Plugin.settings[:ansible_environment_file].shellescape
+        init = <<~EOF
+          #!/bin/sh
+          [ -f #{ansible_env} ] && source #{ansible_env}
+          exec #{command.map(&:shellescape).join(' ')}
+        EOF
+        init_path = File.join(@root, 'init.sh')
+        File.write(init_path, init, { perm: 0o0755 })
```

Comment:
Would it be better to have a static wrapper file somewhere that accepts some standard arguments? I think it saves some disk IO.

Reply:
Yes, that would better. Alternatively we could start a shell and pipe this script to its stdin.

Reply:
Those are common tools also used in exploits. They make tracing what is actually executed harder. I'd be in favor to ship something (like in `/usr/libexec`). This has the additional benefit that we can enforce a different SELinux domain, which means we can prevent it from reading files `foreman-proxy` has access to (like private keys etc).

Reply:
Yes, please make my life harder ;-)
(Read: yay for the process not being able to read random certs etc)

Reply:
I just realized we don't actually have SELinux for foreman-proxy. Still ... ;)